04 - Guides, Workflows, & Courses/Guides/Best Practices and Tips for Theme Development.md

```diff
@@ -0,0 +1,19 @@
+---
+aliases: 
+- 
+tags:
+- seedling
+publish: true
+---
+
+# Best Practices and Tips for Theme Development
+
+This note serves as a collection of tips and best practices for theme development which are too small to be their own guide.
+
+- Name the font-faces in your theme's CSS differently from the one you installed on your system to prevent potential issues with fonts from going unnoticed. 
+    By naming your base64-encoded font-faces (`@font-face`) the same as the font you installed on your system, you will never actually see how the fonts are rendered for other users, since Obsidian will prefer the system font when there are multiple fonts with the same name available (one via base64-encoded CSS, one on the system).
+- Prefer `woff2` over `ttf` as font format, since the former have a far smaller file size.
+- if you ever need more specificity than adding a `body` at the start of your selectors, you can double the targeted class, like so:
+	- `body .timer-start-button.timer-start-button { --button-fill-inner: white; }`
+	- The usecase for this is, for example, when an element uses those `.svelte-xyz` generated classes, which gives the rule more specificity, and just adding a `body` is not enough to override it. you can double or triple the class selector you're targeting, which gives it enough specificity for css to read it as more important.
```

Comment:
It would probably be best to add a disclaimer that using those `.element.svelte` type of classes to a CSS snippet will  make it stop working if there's a change to the plugin, and so it's best to not include the `.svelte` parts themselves (so just the element class), and then add more specificity if required.

Reply:
It's been added, thanks for the note :)